dbms/src/Storages/DeltaMerge/Remote/RNReadTask.cpp

```diff
@@ -0,0 +1,156 @@
+// Copyright 2023 PingCAP, Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <Interpreters/Context.h>
+#include <Interpreters/SharedContexts/Disagg.h>
+#include <Storages/DeltaMerge/ColumnFile/ColumnFileDataProvider.h>
+#include <Storages/DeltaMerge/Remote/RNDataProvider.h>
+#include <Storages/DeltaMerge/Remote/RNLocalPageCache.h>
+#include <Storages/DeltaMerge/Remote/RNReadTask.h>
+#include <Storages/DeltaMerge/Remote/Serializer.h>
+#include <Storages/DeltaMerge/Segment.h>
+
+namespace DB::DM::Remote
+{
+
+RNReadSegmentTaskPtr RNReadSegmentTask::buildFromEstablishResp(
+    const LoggerPtr & log,
+    const Context & db_context,
+    const ScanContextPtr & scan_context,
+    const RemotePb::RemoteSegment & proto,
+    const DisaggTaskId & snapshot_id,
+    StoreID store_id,
+    const String & store_address,
+    KeyspaceID keyspace_id,
+    TableID physical_table_id)
+{
+    RowKeyRange segment_range;
+    {
+        ReadBufferFromString rb(proto.key_range());
+        segment_range = RowKeyRange::deserialize(rb);
+    }
+    RowKeyRanges read_ranges(proto.read_key_ranges_size());
+    for (int i = 0; i < proto.read_key_ranges_size(); ++i)
+    {
+        ReadBufferFromString rb(proto.read_key_ranges(i));
+        read_ranges[i] = RowKeyRange::deserialize(rb);
+    }
+
+    auto dm_context = std::make_shared<DMContext>(
+        db_context,
+        /* path_pool */ nullptr,
+        /* storage_pool */ nullptr,
+        /* min_version */ 0,
```

Comment:
Will `min_version` be changed later? I think we'd better assign the `read_tso` of the reqeust to the  `dm_context.min_version` here

Reply:
Looks like the `min_version` has a misleading name: it is actually used as something like `min_gc_version`. Under this semantic maybe it is fine to not assign it?

Reply:
OK